### PR TITLE
task config: add thread config for mydumper, loader and syncer

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
add new config to make the basic config simple

### What is changed and how it works?
add new config `mydumper-thread`, `loader-thread` and `syncer-thread`

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

**Comment by WangXiangUSTC** [comment]
update sample task config file in https://github.com/pingcap/dm/pull/309/commits/7fd6a2decc0bc6fe5526642f6e2f48a4f0d1e5e0 @csuzhangxc

File: dm/master/task.yaml [hunk 1]
```diff
@@ -35,17 +35,23 @@ mysql-instances:             # one or more source database, config more source d
...
-#      threads: 16
+    # `mydumper-thread` is alias for `threads` in `mydumper` config, and it's priority is high than `threads`
+    #mydumper-thread: 16
+    #mydumper:
+    #  mydumper-path: "./mydumper"
+    #  threads: 16
 
     # `loader-config-name` and `loader` should only set one
     loader-config-name: "global"    # ref `loaders` config
+    # `loader-thread` is alias for `pool-size` in `loader` config, and it's priority is high than `poo-size`
```
Comment (lichunzhu) [inline dm/master/task.yaml]:
```suggestion
    # `loader-thread` is alias for `pool-size` in `loader` config, and it's priority is high than `pool-size`
```

**Comment by ericsyh** [comment]
@WangXiangUSTC @csuzhangxc @lichunzhu 
Shall we delete the previous config `threads`, `pool-size`, `worker-count` in the sample `task.yaml`? 
Since we want to replace them by new configs, there is no need to keep them i think.

File: dm/master/task.yaml [hunk 2]
```diff
@@ -35,17 +35,23 @@ mysql-instances:             # one or more source database, config more source d
...
+    #loader-thread: 32
     #loader:                  # local loader rule
     #  pool-size: 32
     # `syncer-config-name` and `syncer` should only set one
 
     syncer-config-name: "global"    # ref `syncers` config
+    # `syncer-thread` is alias for `worker-count` in `syncer` config, and it's priority is high than `worker-count`
```
Comment (ericsyh) [inline dm/master/task.yaml]:
```suggestion
    # `syncer-thread` is alias for `worker-count` in `syncer` config, and its priority is higher than `worker-count`
```

**Comment by csuzhangxc** [comment]
> @WangXiangUSTC @csuzhangxc @lichunzhu
> Shall we delete the previous config `threads`, `pool-size`, `worker-count` in the sample `task.yaml`?
> Since we want to replace them by new configs, there is no need to keep them i think.

Is it better to provide 2 files, one for basic and another for advanced?

@WangXiangUSTC we also have a task config file under _dm/dm-ansible/conf_ named _task.yaml.example_, I think we should update it too.

**Comment by WangXiangUSTC** [comment]
ok, I will split it to two example config file, one is basic config and another is full config file @ericsyh @csuzhangxc

File: dm/config/task.go [hunk 3]
```diff
@@ -86,10 +86,18 @@ type MySQLInstance struct {
 
 	MydumperConfigName string          `yaml:"mydumper-config-name"`
 	Mydumper           *MydumperConfig `yaml:"mydumper"`
-	LoaderConfigName   string          `yaml:"loader-config-name"`
-	Loader             *LoaderConfig   `yaml:"loader"`
-	SyncerConfigName   string          `yaml:"syncer-config-name"`
-	Syncer             *SyncerConfig   `yaml:"syncer"`
+	// MydumperThread is alias for Threads in MydumperConfig, and it's priority is high than Threads
```
[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply dm/config/task.go]: fixed in https://github.com/pingcap/dm/pull/309/commits/e2a3f360757879a1e51fd6f5a7f1208fd0e39316

**Comment by WangXiangUSTC** [comment]
https://github.com/pingcap/dm/pull/309/commits/db3480d0074914ff1b815a986411192d23600f62
https://github.com/pingcap/dm/pull/309/commits/b0b38512084fb499033e7549225f7255622df65d
done @ericsyh @csuzhangxc

File: dm/master/task_basic.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  port: 4000
+  user: "root"
+  password: ""
+
+mysql-instances:             # one or more source database, config more source database for sharding merge
+  -
+    source-id: "instance118-4306" # unique in all instances, used as id when save checkpoints, configs, etc.
+    black-white-list:  "instance"
+
+    # `mydumper-thread` is alias for `threads` in `mydumper` config, and it's priority is higher than `threads`
```
Comment (csuzhangxc) [inline dm/master/task_basic.yaml]: in _task_basic_ file, give it a meaningful comment?

File: dm/master/task_basic.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    # `syncer-thread` is alias for `worker-count` in `syncer` config, and it's priority is higher than `worker-count`
+    syncer-thread: 16
+
+    
+
+# other common configs shared by all instances
+black-white-list:
+  instance:
+    do-dbs: ["~^test.*", "do"]        # white list of upstream schemas needs to be replicated, regular expression (starts with ~) is supported
+    ignore-dbs: ["mysql", "ignored"]  # black list of upstream schemas needs to be replicated, regular expression (starts with ~) is supported
```
Comment (csuzhangxc) [inline dm/master/task_basic.yaml]: how about only show white list in _task_basic_? @ericsyh

File: dm/master/task_basic.yaml (hunk 4 above)
Comment (WangXiangUSTC) [reply dm/master/task_basic.yaml]: update https://github.com/pingcap/dm/pull/309/commits/1cde56519be4c5da87635a3406aeb6c982a90d21

File: dm/master/task_basic.yaml (hunk 5 above)
Comment (WangXiangUSTC) [reply dm/master/task_basic.yaml]: I think it's ok, update in https://github.com/pingcap/dm/pull/309/commits/1cde56519be4c5da87635a3406aeb6c982a90d21

File: dm/config/task.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (lichunzhu) [reply dm/config/task.go]: @WangXiangUSTC I think `its` is better than `it's`. `it's` usually means `it is` or `it has`.

[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply dm/config/task.go]: you are right, update in https://github.com/pingcap/dm/pull/309/commits/4103d2c029d7c65dd331312c9b1800865e97639d

**Comment by WangXiangUSTC** [comment]
@ericsyh @lichunzhu PTAL again

**Comment by lichunzhu** [comment]
LGTM

**Pull request merged**